src/Packetery/Module/Order/CarrierModal.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Class CarrierModal.
+ *
+ * @package Packetery
+ */
+
+declare( strict_types=1 );
+
+namespace Packetery\Module\Order;
+
+use Packetery\Module;
+use Packetery\Latte\Engine;
+use Packetery\Nette\Forms;
+
+/**
+ * Class CarrierModal.
+ *
+ * @package Packetery
+ */
+class CarrierModal {
+
+	public const MODAL_ID = 'wc-packetery-carrier-modal';
+
+	/**
+	 * Latte engine.
+	 *
+	 * @var Engine
+	 */
+	private $latteEngine;
+
+	/**
+	 * Context resolver.
+	 *
+	 * @var Module\ContextResolver
+	 */
+	private $contextResolver;
+
+	/**
+	 * Order detail common logic.
+	 *
+	 * @var DetailCommonLogic
+	 */
+	private $detailCommonLogic;
+
+	/**
+	 * Carrier repository.
+	 *
+	 * @var Module\Carrier\EntityRepository
+	 */
+	private $carrierRepository;
+
+	/**
+	 * Form factory.
+	 *
+	 * @var Module\FormFactory
+	 */
+	private $formFactory;
+
+	/**
+	 * Order repository.
+	 *
+	 * @var Repository
+	 */
+	private $orderRepository;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Engine                          $latteEngine       Latte engine.
+	 * @param DetailCommonLogic               $detailCommonLogic Detail common logic.
+	 * @param Module\Carrier\EntityRepository $carrierRepository Carrier repository.
+	 * @param Module\FormFactory              $formFactory       Form factory.
+	 * @param Repository                      $orderRepository   Order repository.
+	 */
+	public function __construct(
+		Engine $latteEngine,
+		DetailCommonLogic $detailCommonLogic,
+		Module\Carrier\EntityRepository $carrierRepository,
```

Comment:
Přehlednější by byl import Carrier, aby to bylo Carrier\EntityRepository.
To jen pro info, záleží na tom, kde bude logika, která to potřebuje.

Reply:
Logiku jsem pretahnul jinam a nazev zjednodusil.